lib/jsdom/living/helpers/events.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+const Event = require("../generated/Event");
+const { tryImplForWrapper } = require("../generated/utils");
+
+function createAnEvent(e, eventConstructor = Event, attributes = {}) {
+  return eventConstructor.createImpl(
+    [e, attributes],
+    { isTrusted: attributes.isTrusted !== false }
+  );
+}
+
+function fireAnEvent(e, target, eventConstructor = Event, attributes = {}, legacyTargetOverrideFlag) {
+  const event = createAnEvent(e, eventConstructor, attributes);
+
+  return tryImplForWrapper(target)._dispatch(event, legacyTargetOverrideFlag);
```

Comment:
Do we really need to implForWrapper this? Seems like this should only be called for impls anyways?

Reply:
It seems to be required for when the method is used in `Window.js` and `xmlhttprequest.js`, without additional changes to those files at least.

Reply:
Let's add a comment to explain that it's for those cases; eventually we will be able to clean lots of such tryImplForWrapper() cases up.